src/client.rs

```diff
@@ -1078,10 +1078,29 @@ impl<T: Read + Write> Session<T> {
     /// `EXISTS` response.  If the server does not do so, the client MAY issue a `NOOP` command (or
     /// failing that, a `CHECK` command) after one or more `APPEND` commands.
     pub fn append<S: AsRef<str>, B: AsRef<[u8]>>(&mut self, mailbox: S, content: B) -> Result<()> {
+        self.append_with_flags(mailbox, content, "()")
+    }
+
+    /// The [`APPEND` command](https://tools.ietf.org/html/rfc3501#section-6.3.11) can take
+    /// an optional FLAGS parameter to set the flags on the new message.
+    ///
+    /// > If a flag parenthesized list is specified, the flags SHOULD be set
+    /// > in the resulting message; otherwise, the flag list of the
+    /// > resulting message is set to empty by default.  In either case, the
+    /// > Recent flag is also set.
+    ///
+    /// Behaves like ['Session::append'], except takes a parenthesized list of flags.
+    pub fn append_with_flags<S: AsRef<str>, B: AsRef<[u8]>, C: AsRef<str>>(
+        &mut self,
+        mailbox: S,
+        content: B,
+        flags: C,
```

Comment:
Instead of taking a single string, should we be taking a slice of `&str` instead so the caller does not need to know the exact format? Maybe we can even use the existing `Flag` type?

Reply:
Yes! I think the single string is a bit inconvenient, and a `Vec<Flag>` or `&[Flag]` would be better. What would you prefer? It is only this way for consistency with some of the other APIs, and I would be happy to change it to something nicer to use and harder to use wrong.

Reply:
`&[Flag]` I think is the way to go!